tests/test_config.py

```diff
@@ -328,13 +339,50 @@ def test_invalid_enviroment_variables_config(monkeypatch, capsys):
         _, err = capsys.readouterr()
         assert  "error: unrecognized arguments: --foo" in err
 
+
 def test_cli_overrides_enviroment_variables_module(monkeypatch):
     monkeypatch.setenv("GUNICORN_CMD_ARGS", "--workers=4")
     with AltArgs(["prog_name", "-c", cfg_file(), "--workers", "3"]):
         app = NoConfigApp()
     assert app.cfg.workers == 3
 
 
+def test_wsgi_app_config(capsys):
+    # gunicorn app:app
+    with AltArgs(["prog_name", "app:app"]):
+        app = WSGIApp()
+    assert app.app_uri == "app:app"
+
+    # gunicorn -c test_cfg.py (config file without wsgi_app)
+    with AltArgs(["prog_name", "-c", cfg_file(), "app:app"]):
+        app = WSGIApp()
+    assert app.app_uri == "app:app"
+
+    # gunicorn -c test_cfg1.py (config file with wsgi_app)
+    with AltArgs(["prog_name", "-c", cfg_file1(), "app:app"]):
+        app = WSGIApp()
+    assert app.app_uri == "app:app"
+
+    # gunicorn -c test_cfg1.py (config file with wsgi_app)
+    with AltArgs(["prog_name", "-c", cfg_file1()]):
+        app = WSGIApp()
+    assert app.app_uri == "app1:app1"
+
+    with AltArgs():
```

Comment:
These two could go into a separate test case and use `pytest.mark.parametrize`. See `test_reload` for an example.